### DISCUSSION TITLE
合并uosc单曲循环和列表循环的两个按钮

### DISCUSSION THREAD

**Discussion by Pavomuticus** [hooke007/mpv_PlayKit#217] [Q&A 问答] [answered]
如题，这两个按钮全部点亮的时候不能明确指示当前状态，有没有什么方法能把它们合并起来

**Comment by hooke007** [comment]
为什么不能表示当前状态？假如合并了，才是无法完全显示当前状态

这两本来就不是一个命令实现的功能
https://mpv.io/manual/master/#options-loop-playlist
https://mpv.io/manual/master/#options-loop-file

当你手动切换下一个，它依然遵循列表循环的原则

**Reply by Pavomuticus** [reply]
我其实是这个意思(类似vlc的单按钮循环)
![IMG_20221115_233919.png](https://user-images.githubusercontent.com/89340536/201968461-b51ff304-ffb9-49c0-ac9f-f371896b0228.png)
但理解了单曲循环的优先级要高于顺序播放后，UOSC的逻辑也很好用，谢谢你

**Reply by hooke007** [reply]
这是mpv的逻辑不是uosc的逻辑；
我知道你的意思，但你仔细想就知道你说的恰恰是别的播放器的逻辑漏洞。

这并非是优先级而是针对两个处理对象（文件与列表）：
即使是无限的循环依然有需要跳出循环的时候，比如手动切换，比如片段损坏或者其它脚本或配置预设。
另外循环单文件的次数可以是有限的，单文件的播放次数(循环)达成后，切换到下一个应该遵循什么规则，为什么一定是顺序而不是循环，或者为什么不可以是乱序？

**Reply by Pavomuticus** [reply]
分开来确实更加灵活

**Comment by Pavomuticus** [comment, marked as answer]
粗糙的实现[mpvconfig](https://github.com/Pavomuticus/mpvconfig)
